sdks/python/apache_beam/runners/interactive/interactive_runner.py

```diff
@@ -209,6 +218,62 @@ def visit_transform(self, transform_node):
 
     return main_job_result
 
+  # TODO(victorhc): Move this method somewhere else if performance is impacted
+  # by generating a cluster during runtime.
+  def _create_dataproc_cluster_if_applicable(self, user_pipeline):
+    """ Creates a Dataproc cluster if the provided user_pipeline is running
+    FlinkRunner and no flink_master_url was provided as an option. A cluster
+    is not created when a flink_master_url is detected.
+
+    Example pipeline options to enable automatic Dataproc cluster creation:
+      options = PipelineOptions([
+      '--runner=FlinkRunner',
+      '--project=my-project',
+      '--region=my-region',
+      '--environment_type=DOCKER'
+      ])
+
+    Example pipeline options to skip automatic Dataproc cluster creation:
+      options = PipelineOptions([
+      '--runner=FlinkRunner',
+      '--flink_master=example.internal:41979',
+      '--environment_type=DOCKER'
+      ])
+    """
+    from apache_beam.options.pipeline_options import FlinkRunnerOptions
+    flink_master = user_pipeline.options.view_as(
+        FlinkRunnerOptions).flink_master
+    if flink_master == '[auto]':
+      from apache_beam.runners.portability.flink_runner import FlinkRunner
+      if isinstance(self._underlying_runner, FlinkRunner):
```

Comment:
Can this be

```Python
# Only consider this logic when both below 2 conditions apply.
if isinstance(self._underlying_runner, 
    FlinkRunner) and clusters.dataproc_cluster_managers.get(
    str(id(user_pipeline)), None) is None:
  if flink_master == '[auto]':
    # build_metadata from options
  elif flink_master in ie.current_env().clusters.master_urls:
    # get metadata from managed master_urls
    cluster_metadata = ie.current_env().clusters.master_urls.get(flink_master)
  # Else noop, no need to log anything because we allow a master_url (not managed by us) provided by the user.
  if cluster_metadata:
    # create the cluster_manager and populate dicts in the clusters instance.
```

Reply:
This flow is much nicer, thanks! I've changed the format and added additional comments for clarity.